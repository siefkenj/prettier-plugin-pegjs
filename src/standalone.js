//
// This file contains a standalone parser which includes Prettier
//
import Prettier from "prettier/standalone";
import * as prettierPluginPegjs from "./prettier-plugin-pegjs";

// Because we're importing the standalone version of prettier, we also have
// to import the specific plugins we want

import babelPlugin from "prettier/parser-babel";

/**
 * Format `source` LaTeX code using Prettier to format/render
 * the code.
 *
 * @export
 * @param {string} [source=""] - code to be formatted
 * @param {*} [options={}] - Prettier options object (you can set `printWidth` here)
 * @returns {string} formatted code
 */
function printPrettier(source = "", options = {}) {
    // Load the prettier and babel plugins, but also allow
    // other plugins to be passed in.
    const plugins = options.plugins || [];
    plugins.push(prettierPluginPegjs, babelPlugin);

    return Prettier.format(source, {
        printWidth: 80,
        useTabs: false,
        tabWidth: 2,
        ...options,
        parser: "pegjs-parser",
        plugins,
    });
}

export { Prettier, printPrettier, prettierPluginPegjs };
