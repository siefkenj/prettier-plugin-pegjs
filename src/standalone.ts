//
// This file contains a standalone parser which includes Prettier
//
import * as Prettier from "prettier/standalone";
import * as prettierPluginPegjs from "./prettier-plugin-pegjs";

// Because we're importing the standalone version of prettier, we also have
// to import the specific plugins we want

import * as babelPlugin from "prettier/parser-babel";
import type { Options, Plugin } from "prettier";

/**
 * Format `source` LaTeX code using Prettier to format/render
 * the code.
 *
 * @export
 * @param [source=""] - code to be formatted
 * @param [options={}] - Prettier options object (you can set `printWidth` here)
 * @returns formatted code
 */
function printPrettier(
    source = "",
    options: Options & { actionParser?: string } = {}
) {
    // Load the prettier and babel plugins, but also allow
    // other plugins to be passed in.
    const plugins = options.plugins || [];
    plugins.push(prettierPluginPegjs as unknown as Plugin, babelPlugin);

    return Prettier.format(source, {
        printWidth: 80,
        useTabs: false,
        tabWidth: 2,
        ...options,
        parser: "pegjs-parser",
        plugins,
    });
}

export { Prettier, printPrettier, prettierPluginPegjs };

export const x=5