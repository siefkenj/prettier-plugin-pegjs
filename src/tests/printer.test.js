import util from "util";

import { printPrettier } from "../standalone";

/* eslint-env jest */

// Make console.log pretty-print by default
export const origLog = console.log;
console.log = (...args) => {
    origLog(...args.map((x) => util.inspect(x, false, 10, true)));
};

describe("Printer", () => {
    it("Prints grammars without actions", () => {
        const sources = [
            "Rule = a/b/c",
            "Rule = a",
            "Rule = [a-zA-Z]",
            "Rule = a *",
            "Rule = (a/b)?",
            "Rule = (a/b/c)   ?",
            "Rule = (a/(b/c)+)   ?",
            "Rule = $(a/(b/c)+)   ?",
            "Rule = $(a/(b/c)+)   ?\n OtherRule= Rule & 'q'",
        ];

        for (const src of sources) {
            const formatted = printPrettier(src, { printWidth: 80 });
            expect(formatted).toMatchSnapshot();
        }
    });
    it("Prints grammars with actions", () => {
        const sources = [
            "start = (a:'a') &{ return a === 'a'; }",
            "start = (a:'a')? &{ return a === 'a'; }",
            "start = (a:'a')* &{ return a === 'a'; }",
            "start = (a:'a')+ &{ return a === 'a'; }",
            "start = $(a:'a') &{ return a === 'a'; }",
            "start = &(a:'a') 'a' &{ return a === 'a'; }",
            "start = !(a:'a') 'b' &{ return a === 'a'; }",
            "start = b:(a:'a') &{ return a === 'a'; }",
            "start = ('a' b:'b' 'c') &{ return b === 'b'; }",
            "start = (a:'a' { return a; }) &{ return a === 'a'; }",
            "start = ('a' / b:'b' / 'c') &{ return b === 'b'; }",
            "start = (a:'a') !{ return a !== 'a'; }",
            "start = (a:'a')? !{ return a !== 'a'; }",
            "start = (a:'a')* !{ return a !== 'a'; }",
            "start = (a:'a')+ !{ return a !== 'a'; }",
            "start = $(a:'a') !{ return a !== 'a'; }",
            "start = &(a:'a') 'a' !{ return a !== 'a'; }",
            "start = !(a:'a') 'b' !{ return a !== 'a'; }",
            "start = b:(a:'a') !{ return a !== 'a'; }",
            "start = ('a' b:'b' 'c') !{ return b !== 'b'; }",
            "start = (a:'a' { return a; }) !{ return a !== 'a'; }",
            "start = ('a' / b:'b' / 'c') !{ return b !== 'b'; }",
            "start = (a:'a') { return a; }",
            "start = (a:'a')? { return a; }",
            "start = (a:'a')* { return a; }",
            "start = (a:'a')+ { return a; }",
            "start = $(a:'a') { return a; }",
            "start = &(a:'a') 'a' { return a; }",
            "start = !(a:'a') 'b' { return a; }",
            "start = b:(a:'a') { return a; }",
            "start = ('a' b:'b' 'c') { return b; }",
            "start = (a:'a' { return a; }) { return a; }",
            "start = ('a' / b:'b' / 'c') { return b; }",
        ];

        for (const src of sources) {
            const formatted = printPrettier(src, { printWidth: 80 });
            expect(formatted).toMatchSnapshot();
        }
    });
    it("Prints grammars with initializer", () => {
        const sources = [
            "{console.log('initializing')}; Rule = a/b/c",
            "{console.log('initializing')}\n\n Rule = a/b/c",
            "{{console.log('initializing global')}}\n\n Rule = a/b/c",
            "{{console.log('initializing global')}}; Rule = a/b/c",
            "{{console.log('initializing global')}}\n\n {console.log('initializing local')}\n\n Rule = a/b/c",
            "{{console.log('initializing global')}}; {console.log('initializing local')}; Rule = a/b/c",
        ];

        for (const src of sources) {
            const formatted = printPrettier(src, { printWidth: 80 });
            expect(formatted).toMatchSnapshot();
        }
    });
    it("Prints grammars with comments", () => {
        const sources = [
            "start = // a comment\n a / b",
            "start = a // a comment\n / b",
            "start = a \n// a comment\n / b",
            "start = a / \n// a comment\n  b",
            "// a comment\n start = a / b",
            "start /*inline comment*/= a / b",
            "start = a / /*inline comment*/ b",
            "start = a / x /*inline comment*/ b",
            'start /*inline comment*/ "Start Label"= a / b',
        ];

        for (const src of sources) {
            const formatted = printPrettier(src, { printWidth: 80 });
            expect(formatted).toMatchSnapshot();
        }
    });
    it("Issue 18 nested optional concat", ()=>{
        const sources = [
            "start = ($\"x\"+)?",
            "start = $(\"x\"+)?",
        ];

        for (const src of sources) {
            const formatted = printPrettier(src, { printWidth: 80 });
            expect(formatted).toMatchSnapshot();
        }
    })
});
