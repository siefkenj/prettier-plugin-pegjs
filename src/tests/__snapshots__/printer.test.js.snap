// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Printer Issue 18 nested optional concat 1`] = `
"start = ($"x"+)?
"
`;

exports[`Printer Issue 18 nested optional concat 2`] = `
"start = $("x"+)?
"
`;

exports[`Printer Prints grammars with actions 1`] = `
"start = (a:"a") & { return a === "a"; }
"
`;

exports[`Printer Prints grammars with actions 2`] = `
"start = (a:"a")? & { return a === "a"; }
"
`;

exports[`Printer Prints grammars with actions 3`] = `
"start = (a:"a")* & { return a === "a"; }
"
`;

exports[`Printer Prints grammars with actions 4`] = `
"start = (a:"a")+ & { return a === "a"; }
"
`;

exports[`Printer Prints grammars with actions 5`] = `
"start = $(a:"a") & { return a === "a"; }
"
`;

exports[`Printer Prints grammars with actions 6`] = `
"start = &(a:"a") "a" & { return a === "a"; }
"
`;

exports[`Printer Prints grammars with actions 7`] = `
"start = !(a:"a") "b" & { return a === "a"; }
"
`;

exports[`Printer Prints grammars with actions 8`] = `
"start = b:(a:"a") & { return a === "a"; }
"
`;

exports[`Printer Prints grammars with actions 9`] = `
"start = ("a" b:"b" "c") & { return b === "b"; }
"
`;

exports[`Printer Prints grammars with actions 10`] = `
"start = (a:"a" { return a; }) & { return a === "a"; }
"
`;

exports[`Printer Prints grammars with actions 11`] = `
"start = ("a" / b:"b" / "c") & { return b === "b"; }
"
`;

exports[`Printer Prints grammars with actions 12`] = `
"start = (a:"a") ! { return a !== "a"; }
"
`;

exports[`Printer Prints grammars with actions 13`] = `
"start = (a:"a")? ! { return a !== "a"; }
"
`;

exports[`Printer Prints grammars with actions 14`] = `
"start = (a:"a")* ! { return a !== "a"; }
"
`;

exports[`Printer Prints grammars with actions 15`] = `
"start = (a:"a")+ ! { return a !== "a"; }
"
`;

exports[`Printer Prints grammars with actions 16`] = `
"start = $(a:"a") ! { return a !== "a"; }
"
`;

exports[`Printer Prints grammars with actions 17`] = `
"start = &(a:"a") "a" ! { return a !== "a"; }
"
`;

exports[`Printer Prints grammars with actions 18`] = `
"start = !(a:"a") "b" ! { return a !== "a"; }
"
`;

exports[`Printer Prints grammars with actions 19`] = `
"start = b:(a:"a") ! { return a !== "a"; }
"
`;

exports[`Printer Prints grammars with actions 20`] = `
"start = ("a" b:"b" "c") ! { return b !== "b"; }
"
`;

exports[`Printer Prints grammars with actions 21`] = `
"start = (a:"a" { return a; }) ! { return a !== "a"; }
"
`;

exports[`Printer Prints grammars with actions 22`] = `
"start = ("a" / b:"b" / "c") ! { return b !== "b"; }
"
`;

exports[`Printer Prints grammars with actions 23`] = `
"start = (a:"a") { return a; }
"
`;

exports[`Printer Prints grammars with actions 24`] = `
"start = (a:"a")? { return a; }
"
`;

exports[`Printer Prints grammars with actions 25`] = `
"start = (a:"a")* { return a; }
"
`;

exports[`Printer Prints grammars with actions 26`] = `
"start = (a:"a")+ { return a; }
"
`;

exports[`Printer Prints grammars with actions 27`] = `
"start = $(a:"a") { return a; }
"
`;

exports[`Printer Prints grammars with actions 28`] = `
"start = &(a:"a") "a" { return a; }
"
`;

exports[`Printer Prints grammars with actions 29`] = `
"start = !(a:"a") "b" { return a; }
"
`;

exports[`Printer Prints grammars with actions 30`] = `
"start = b:(a:"a") { return a; }
"
`;

exports[`Printer Prints grammars with actions 31`] = `
"start = ("a" b:"b" "c") { return b; }
"
`;

exports[`Printer Prints grammars with actions 32`] = `
"start = (a:"a" { return a; }) { return a; }
"
`;

exports[`Printer Prints grammars with actions 33`] = `
"start = ("a" / b:"b" / "c") { return b; }
"
`;

exports[`Printer Prints grammars with comments 1`] = `
"start
  // a comment
  = a
  / b
"
`;

exports[`Printer Prints grammars with comments 2`] = `
"start
  = a // a comment
  / b
"
`;

exports[`Printer Prints grammars with comments 3`] = `
"start
  = a
  // a comment
  / b
"
`;

exports[`Printer Prints grammars with comments 4`] = `
"start
  = a
  / b
  // a comment
"
`;

exports[`Printer Prints grammars with comments 5`] = `
"// a comment
start
  = a
  / b
"
`;

exports[`Printer Prints grammars with comments 6`] = `
"start
  /*inline comment*/ = a
  / b
"
`;

exports[`Printer Prints grammars with comments 7`] = `
"start
  = a
  / /*inline comment*/ b
"
`;

exports[`Printer Prints grammars with comments 8`] = `
"start
  = a
  / x /*inline comment*/ b
"
`;

exports[`Printer Prints grammars with comments 9`] = `
"start /*inline comment*/ "Start Label"
  = a
  / b
"
`;

exports[`Printer Prints grammars with initializer 1`] = `
"{ console.log("initializing"); }

Rule
  = a
  / b
  / c
"
`;

exports[`Printer Prints grammars with initializer 2`] = `
"{ console.log("initializing"); }

Rule
  = a
  / b
  / c
"
`;

exports[`Printer Prints grammars with initializer 3`] = `
"{{ console.log("initializing global"); }}

Rule
  = a
  / b
  / c
"
`;

exports[`Printer Prints grammars with initializer 4`] = `
"{{ console.log("initializing global"); }}

Rule
  = a
  / b
  / c
"
`;

exports[`Printer Prints grammars with initializer 5`] = `
"{{ console.log("initializing global"); }}

{ console.log("initializing local"); }

Rule
  = a
  / b
  / c
"
`;

exports[`Printer Prints grammars with initializer 6`] = `
"{{ console.log("initializing global"); }}

{ console.log("initializing local"); }

Rule
  = a
  / b
  / c
"
`;

exports[`Printer Prints grammars without actions 1`] = `
"Rule
  = a
  / b
  / c
"
`;

exports[`Printer Prints grammars without actions 2`] = `
"Rule = a
"
`;

exports[`Printer Prints grammars without actions 3`] = `
"Rule = [a-zA-Z]
"
`;

exports[`Printer Prints grammars without actions 4`] = `
"Rule = a*
"
`;

exports[`Printer Prints grammars without actions 5`] = `
"Rule = (a / b)?
"
`;

exports[`Printer Prints grammars without actions 6`] = `
"Rule = (a / b / c)?
"
`;

exports[`Printer Prints grammars without actions 7`] = `
"Rule = (a / (b / c)+)?
"
`;

exports[`Printer Prints grammars without actions 8`] = `
"Rule = $(a / (b / c)+)?
"
`;

exports[`Printer Prints grammars without actions 9`] = `
"Rule = $(a / (b / c)+)?

OtherRule = Rule &"q"
"
`;
